manager/integration/tests/test_engine_upgrade.py

```diff
@@ -683,10 +683,11 @@ def test_engine_live_upgrade_with_intensive_data_writing(client, core_api, volum
     for replica in volume.replicas:
         assert replica.engineImage == engine_upgrade_image
 
-    # The reattached volume should be state `healthy` rather than`degraded`.
     create_and_wait_pod(core_api, pod)
-    volume = client.by_id_volume(volume_name)
-    assert volume.robustness == "healthy"
+    common.wait_for_volume_degraded(client, volume_name)
```

Comment:
There is a problem with this approach. The degraded/rebuild start/rebuild complete might happen quickly, so it's hard to detect those intermediate states. And if we miss them, then the test case will fail.

We can wait for the volume to become healthy instead.

Reply:
Since the replica rebuilds sequentially and this one has 800MiB data and 100Mib ongoing, and this sometimes fails for me when the test failed by timed out during the rebuilding of the last replica.  To include degraded/rebuild start/rebuild complete, the test result can have more info about which stage was causing the issues. 

And since some combinations implemented in other tests and so I assumed this shouldn't be a problem. 
https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/test_ha.py#L1225-L1234
https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/test_ha.py#L1966-L1967

Reply:
With enough data, yes this makes sense.